Update CONTRIBUTING.md, add MAINTAINERS.md (#309)
theupdateframework/go-tuf: docs/MAINTAINERS.md

# go-tuf maintainer guidelines

These are expectations for the [MAINTAINERS](MAINTAINERS) of go-tuf; if you are not able to meet these requirements, please remove yourself from the list of maintainers.

## Process

Speedy communication makes contributors happy!

- You should get notifications for all activity in this repository (using the "Watch" feature) and quickly triage each issue/PR as it comes in.
  - (non-draft) PRs should have assigned reviewers.
  - Important bugs and questions should have assignees.
- If you are assigned to review a PR, please try to *acknowledge* it within one business day (no need if you are OOO).
- Please review all PRs within five business days (of course, it's okay if you're OOO).
- Please use the review checklist below.

> - We should make sure there's a reviewer for every PR with tests passing within 

Reviewer: ```suggestion
- We should make sure there's an assigned reviewer for every non-draft PR which has passing tests.
```
